substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs

```diff
@@ -65,41 +118,28 @@ type UncheckedSignaturePayload<Address, Signature, Extra> = (Address, Signature,
 /// This can be checked using [`Checkable`], yielding a [`CheckedExtrinsic`], which is the
 /// counterpart of this type after its signature (and other non-negotiable validity checks) have
 /// passed.
-#[derive(PartialEq, Eq, Clone)]
-pub struct UncheckedExtrinsic<Address, Call, Signature, Extra>
-where
-	Extra: SignedExtension,
-{
-	/// The signature, address, number of extrinsics have come before from the same signer and an
-	/// era describing the longevity of this transaction, if this is a signed extrinsic.
-	///
-	/// `None` if it is unsigned or an inherent.
-	pub signature: Option<UncheckedSignaturePayload<Address, Signature, Extra>>,
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct UncheckedExtrinsic<Address, Call, Signature, Extension> {
```

Comment:
Are the docs for this type up to date still? Especially in regard to extrinsic types etc. 

We reference this type + the diagram linked to it a lot as the source of truth for "types of extrinsic"

Reply:
This is still true for old-school extrinsics but incomplete given the general transaction type. This doc comment along with the mermaid file need to be updated. Opened #3592 for this.